substrate/frame/election-provider-multi-block/src/lib.rs

```diff
@@ -663,7 +663,7 @@ pub mod pallet {
 				},
 				Phase::Snapshot(x) => {
 					// create voter snapshot
-					Self::create_voters_snapshot_paged(x).unwrap();
+					Self::create_voters_snapshot_paged(x).defensive_unwrap_or_default();
```

Comment:
Were we doing unwrap in the on-chain code 😮

Reply:
I think this is what you mentioned in our chat yesterday. In these non-extrinsic code paths, we can’t return an error, but we also don’t want to silently fail. The defensive check will emit an error log, and it might also be good to emit an `error` event here for visibility.